libcalico-go/lib/backend/k8s/resources/ipam_affinity.go

```diff
@@ -138,6 +138,10 @@ func (c blockAffinityClient) parseKey(k model.Key) (name, cidr, host string) {
 // toV3 takes the given v1 KVPair and converts it into a v3 representation, suitable
 // for writing as a CRD to the Kubernetes API.
 func (c blockAffinityClient) toV3(kvpv1 *model.KVPair) *model.KVPair {
+	if !isV1Key(kvp1.Key) {
```

Comment:
I say remove this code, and let's make sure we never call this function with a v3 object.